pallets/sponsorship/src/lib.rs

```diff
@@ -430,19 +432,29 @@ pub mod pallet {
 		/// are needed during pre and post dispatching this call.
 		#[pallet::call_index(4)]
 		#[pallet::weight({
-		let dispatch_info = call.get_dispatch_info();
-		(dispatch_info.weight + < T as Config >::WeightInfo::pre_sponsor() + < T as Config >::WeightInfo::post_sponsor() + T::DbWeight::get().reads_writes(2, 2), dispatch_info.class, Pays::No)
-		})]
+		let dispatch_infos = calls.iter().map(|call| call.get_dispatch_info()).collect::<Vec<_>>();
```

Comment:
Compare to https://github.com/paritytech/substrate/blob/033d4e86cc7eff0066cd376b9375f815761d653c/frame/utility/src/lib.rs#L305C1-L322C6